lib/CL/devices/level0/level0-compilation.cc

```diff
@@ -0,0 +1,674 @@
+/* level0-compilation.cc - multithreaded compilation for LevelZero Compute API devices.
+
+   Copyright (c) 2022-2023 Michal Babej / Intel Finland Oy
+
+   Permission is hereby granted, free of charge, to any person obtaining a copy
+   of this software and associated documentation files (the "Software"), to
+   deal in the Software without restriction, including without limitation the
+   rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+   sell copies of the Software, and to permit persons to whom the Software is
+   furnished to do so, subject to the following conditions:
+
+   The above copyright notice and this permission notice shall be included in
+   all copies or substantial portions of the Software.
+
+   THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+   IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+   FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+   AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+   LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+   FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+   IN THE SOFTWARE.
+*/
+
+#include "level0-compilation.hh"
+
+#include "common.h"
+#include "common_driver.h"
+#include "devices.h"
+#include "pocl_cache.h"
+#include "pocl_cl.h"
+#include "pocl_file_util.h"
+#include "pocl_timing.h"
+#include "pocl_util.h"
+
+#include <algorithm>
+#include <atomic>
+#include <cassert>
+#include <cstring>
+#include <iomanip>
+#include <sstream>
+
+// ***********************************************************************
```

Comment:
Pls describe what this class does and what are its special features.

Reply:
i've added doxygen documentation for all classes in the `level0-compilation.hh` header